packages/tpetra/core/src/Tpetra_CrsMatrix_def.hpp

```diff
@@ -8200,6 +8200,8 @@ namespace Tpetra {
         bool mismatch = source_vals != target_vals;
         immRedErr = MPI_Iallreduce(&mismatch,&reduced_mismatch,1,MPI::BOOL,MPI_LOR,rawComm,&rawRequest);
```

Comment:
Trilinos does not test any more with `MPI_VERSION < 3`, so I'm OK with leaving this `MPI_Iallreduce` in place.  Another option would be to use the existing `Tpetra::Details::iallreduce` function, that works regardless of `MPI_VERSION`.  (It uses `MPI_Iallreduce` for `MPI_VERSION >= 3`, and does something else otherwise that is still nonblocking.)  Let's leave that for later, though.